docs/mqtt.md

```diff
@@ -594,6 +594,31 @@ to reconnect to switch to a new virtual host.
 is considered more specific than the port-to-vhost mapping with the `mqtt_port_to_vhost_mapping`
 global parameter and so takes precedence over it.
 
+#### Use client_id from client certificate to authenticate 
+
+By the default, the plugin passes the `client_id` it received from the MQTT connection to the configured authentication backends as a credential. However, it is possible to extract the `client_id` from the certificate in a very similar way as we extract the username. 
+
+First, you must configure which certificate's attribute has the `client_id` by setting the configuration variable `mqtt.ssl_cert_client_id_from`. The allowed values are:
+- `distinguished_name`, this is the full DN, or distinguished name, of the certificate
+- `common_name`, this is just the CN part of the distinguished name
+- `subject_alternative_name` or `subject_alt_name`, here the `client_id` is specified in certificate's extension which allows to specify further alternative names which can be of several types.
+
+If you set `mqtt.ssl_cert_client_id_from` to `subject_alternative_name` or `subject_alt_name`, you should configure `mqtt.ssl_cert_client_id_san_type` which contains the `client_id`, otherwise it defaults to `dns`, its type which can be:
```

Comment:
?? "you should" or "you must" in the 1st sentence? Do they have to do it, "you should" is too vague, if you are going to use "should", I would additionally explaining why they should, thanks

Reply:
its type which can be:

??? should that be a new sentence written as "It's type can be: "

Reply:
I chose `should` because if you do not set it, it is defaulted to `dns`. If that is what the user wants, then the user does not need to configure it.

Reply:
then I would say "If you want `mqtt.ssl_cert_client_id_from` to  be set to`subject_alternative_name` or `subject_alt_name`, then you must.....

Reply:
At the end, i have gone for "you can configure ".. if the user wants to use `dns` as the SAN type, it is not required to set it `mqtt.ssl_cert_client_id_san_type` to `dns` as that is the default behaviour. Are ok ?